src/Quaternion.java

```diff
@@ -83,6 +135,84 @@ public Quaternion multiply(double scalar) {
     }
 
     /**
+     * Method to conjugate a Quaternion.
+     *
+     * @return Returns a new Quaternion with the conjugate.
+     */
+    public Quaternion conjugate() {
+        return new Quaternion(
+                this.getR(),
+                -this.getI(),
+                -this.getJ(),
+                -this.getK());
+    }
+
+    /**
+     * Method to conjugate a Quaternion using multiplications only.
+     *
+     * @return Returns a new Quaternion with the conjugate, but using the formula
+     * q* = -(1/2)*(q + iqi + jqj + kqk).
+     */
+    public Quaternion conjugateMultiplying() {
+        Quaternion i = new Quaternion(0, 1, 0, 0);
+        Quaternion j = new Quaternion(0, 0, 1, 0);
+        Quaternion k = new Quaternion(0, 0, 0, 1);
+
+        Quaternion result = this;
+        result = result.add(i.multiply(this.multiply(i)));
+        result = result.add(j.multiply(this.multiply(j)));
+        result = result.add(k.multiply(this.multiply(k)));
+        return result.multiply(-0.5);
+    }
+
+    /**
+     * Method to calculate the norm of a Quaternion.
+     *
+     * @return Returns the norm of the quaternion.
+     */
+    public double norm() {
+        return Math.sqrt(this.getR() * this.getR() +
+                this.getI() * this.getI() +
+                this.getJ() * this.getJ() +
+                this.getK() * this.getK());
+    }
+
+    /**
+     * Method to normalize a Quaternion.
+     *
+     * @return Returns a new Quaternion with unit norm, un the same direction.
+     */
+    public Quaternion normalize() {
+
+        double norm = this.norm();
+
+        if (norm < MIN_TOLERANCE) {
+            throw new IllegalArgumentException("Trying to normalize a zero norm quaternion");
```

Comment:
+1 on the exception, not sure though that `IllegalArgumentException` is the one to be throwing now as this method doesn't have any arguments.